### PR TITLE
Remove commontypes and automate CRD conversion functions generation

### PR DESCRIPTION
**Pull request by Annopaolo**
- Shared common types between different API versions do not follow k8s best practices and can become an issue later on (e.g. API change, automatic doc generation, ...). Move common types back into v1alpha1 and v1alpha2 CRDs, partially reverting #185.
- Astarte CRDs have too many fields to write conversions by hand. Use the `conversion-gen` tool to generate conversion functions automatically. Include the code generation in `make generate`.
- Update the conversion webhooks for  Astarte, AVI, Flow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,13 +66,13 @@ Astarte's Documentation.
...
-| v1.19.x  | :white_check_mark: | :white_check_mark: |
-| v1.20.x  | :white_check_mark: | :white_check_mark: |
-| v1.21.x  | :white_check_mark: | :white_check_mark: |
+| Kubernetes Version | Supported          | Tested by CI       |
+| ------------------ | ------------------ | ------------------ |
+| v1.17.x            | :x:                | :x:                |
+| v1.18.x            | :x:                | :x:                |
+| v1.19.x            | :white_check_mark: | :white_check_mark: |
+| v1.20.x            | :white_check_mark: | :white_check_mark: |
+| v1.21.x            | :white_check_mark: | :white_check_mark: |
```
Comment (matt-mazzucato) [inline README.md:75]: Are these changes really needed?

File: Makefile [hunk 2]
```diff
@@ -168,6 +170,22 @@ else
...
 
+.PHONY: conversion-gen
+conversion-gen: ## Download conversion-gen locally if necessary.
+ifeq (, $(shell which conversion-gen))
+	@{ \
+	set -e ;\
+	CONVERSION_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONVERSION_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get k8s.io/code-generator/cmd/conversion-gen ;\
```
Comment (matt-mazzucato) [inline Makefile:181]: Maybe set the version as we do for other tools.

**Review by matt-mazzucato — COMMENTED** [review]
Just a couple of minor comments. The overall PR looks good.

**Review by matt-mazzucato — APPROVED** [review]

**Review by rbino — APPROVED** [review]

**Review by bettio — APPROVED** [review]
